consumer/pdata/generated_resource.go

```diff
@@ -66,14 +63,16 @@ func (ms Resource) Attributes() AttributeMap {
 	return newAttributeMap(&(*ms.orig).Attributes)
 }
 
+// IsEmpty returns true if the underlying data are equivalent with an empty message.
+func (ms Resource) IsEmpty() bool {
+	return ms.Attributes().IsEmpty()
+}
+
 // CopyTo copies all properties from the current struct to the dest.
 func (ms Resource) CopyTo(dest Resource) {
-	if ms.IsNil() {
-		*dest.orig = nil
+	if ms.IsEmpty() {
+		*dest.orig = emptyValueResource
 		return
 	}
```

Comment:
Why is this needed? Isn't it sufficient to do `CopyTo` for all fields?

Reply:
Removed.